src/nft/components/details/detailsV2/TabbedComponent.tsx

```diff
@@ -0,0 +1,62 @@
+import Row from 'components/Row'
+import { useState } from 'react'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+import { containerStyles } from './shared'
+
+const TabbedComponentContainer = styled.div`
+  height: 288px;
+
+  ${containerStyles}
+`
+
+const TabsRow = styled(Row)`
+  gap: 32px;
+  margin-bottom: 12px;
+  width: 100;
+`
+
+const Tab = styled(ThemedText.SubHeader)<{ isActive: boolean; numTabs: number }>`
+  color: ${({ theme, isActive }) => (isActive ? theme.textPrimary : theme.textTertiary)};
+  line-height: 24px;
+  cursor: ${({ numTabs }) => (numTabs > 1 ? 'pointer' : 'default')};
+
+  &:hover {
+    opacity: ${({ numTabs, theme }) => numTabs > 1 && theme.opacity.hover}};
+  }
+`
+
+export interface Tab {
+  title: React.ReactNode
+  key: string
+  content: JSX.Element
+}
+
+interface TabbedComponentProps {
+  tabs: Tab[]
+  defaultTab: Tab
+  style?: React.CSSProperties
+}
+
+export const TabbedComponent = ({ tabs, defaultTab, style }: TabbedComponentProps) => {
+  const [activeTab, setActiveTab] = useState(defaultTab.key)
```

Comment:
why not use array index?

Reply:
By using the `key` we can know which is the active tab without outside reference to the entire `tabs` array. I think this is a better object oriented pattern than relying on the tab's position in the array.